pkg/storage/elasticsearch_dependencies.go

```diff
@@ -53,22 +74,17 @@ func elasticsearchDependencies(jaeger *v1.Jaeger) []batchv1.Job {
 	return []batchv1.Job{job}
 }
 
-func envVars(spec v1.Options) []corev1.EnvVar {
-	var envs []corev1.EnvVar
-	if val, ok := spec.Map()["es.index-prefix"]; ok {
-		envs = append(envs, corev1.EnvVar{Name: "INDEX_PREFIX", Value: val})
-	}
-	if val, ok := spec.Map()["es.num-shards"]; ok {
-		envs = append(envs, corev1.EnvVar{Name: "SHARDS", Value: val})
-	}
-	if val, ok := spec.Map()["es.num-replicas"]; ok {
-		envs = append(envs, corev1.EnvVar{Name: "REPLICAS", Value: val})
-	}
-	if val, ok := spec.Map()["es.username"]; ok {
-		envs = append(envs, corev1.EnvVar{Name: "ES_USERNAME", Value: val})
-	}
-	if val, ok := spec.Map()["es.password"]; ok {
-		envs = append(envs, corev1.EnvVar{Name: "ES_PASSWORD", Value: val})
+func envVars(opts v1.Options) []corev1.EnvVar {
+	var envs = cronjob.EsScriptEnvVars(opts)
+	//envs := []corev1.EnvVar{}
+	options := opts.Map()
+	for k, v := range map[string]string{
+		"es.num-shards":   "SHARDS",
```

Comment:
Why are the other options/env vars not copied? index prefix, username and password?

I see from the test they are still env vars, so what changed?

Reply:
They are copied in a function from `cronjob` package just right above. `var envs = cronjob.EsScriptEnvVars(opts)`